test/apple/macos-system-helper.test.ts

```diff
@@ -0,0 +1,163 @@
+//
+// Testing an implementation, i.e. MacOSSystemHelpers.findXcodeAppPath(), by using the same approach in the unit test
+// is bad practice, but it's the only way because the path to the Xcode.app can be different on different machines.
+//
+// While creating these tests we ensured that the implementation in the tests is correct by comparing.
+// We must not change the implementation in the test code!
+
+describe('MacOSSystemHelpers', () => {
+  afterEach(() => {
+    vi.clearAllMocks();
+    vi.restoreAllMocks();
+  });
+
+  describe('findSDKRootDirectoryPath', () => {
+    test.runIf(process.platform === 'darwin')(
+      'should return the SDK root directory path',
+      () => {
+        // -- Arrange --
+        let sdkPath: string;
+        if (process.env.SDK_PATH) {
+          // It is expected that the SDK_PATH environment variable is set in CI.
+          sdkPath = process.env.SDK_PATH;
+        } else {
+          // This is a fallback implementation for local development.
+          // It mostly verifies that the implementation of findSDKRootDirectoryPath() is still unchanged,
+          // not that the path is correct.
+          sdkPath = execSync('xcrun --show-sdk-path', {
+            encoding: 'utf8',
+          }).trim();
+        }
+
+        // -- Act --
+        const sdkRootDirectoryPath =
+          MacOSSystemHelpers.findSDKRootDirectoryPath();
+
+        // -- Assert --
+        expect(sdkRootDirectoryPath).toBe(sdkPath);
+      },
+    );
+
+    test.runIf(process.platform !== 'darwin')(
+      'should return undefined on non-macOS platforms',
+      () => {
+        // The purpose of this test is to verify that the implementation of findSDKRootDirectoryPath() is still unchanged
+
+        // -- Act --
+        const sdkRootDirectoryPath =
+          MacOSSystemHelpers.findSDKRootDirectoryPath();
+
+        // -- Assert --
+        expect(sdkRootDirectoryPath).toBeUndefined();
+      },
+    );
+  });
+
+  describe('findDeveloperDirectoryPath', () => {
+    test.runIf(process.platform === 'darwin')(
+      'should return the developer directory path',
+      () => {
+        // -- Arrange --
+        let xcodeAppPath: string;
+        if (process.env.XCODE_DEVELOPER_DIR_PATH) {
+          // It is expected that the XCODE_DEVELOPER_DIR_PATH environment variable is set in CI.
+          xcodeAppPath = process.env.XCODE_DEVELOPER_DIR_PATH;
+        } else {
+          // This is a fallback implementation for local development.
+          // It mostly verifies that the implementation of findDeveloperDirectoryPath() is still unchanged,
+          // not that the path is correct.
+          xcodeAppPath = execSync('xcode-select --print-path', {
+            encoding: 'utf8',
+          }).trim();
+        }
+
+        // -- Act --
+        const developerDirectoryPath =
+          MacOSSystemHelpers.findDeveloperDirectoryPath();
+
+        // -- Assert --
+        expect(developerDirectoryPath).toBe(xcodeAppPath);
```

Comment:
`m`: If I'm not mistaken, it's not a problem that the `MacOSSystemHelpers.findDeveloperDirectoryPath` returns a different result in CI, as pointed out here: https://github.com/getsentry/sentry-wizard/pull/904#discussion_r2048733766. So instead of passing in the path via a env variable, wouldn't it be sufficient to adapt the assertion to ensure it's either the local variation or the CI variation?

Reply:
Yes, we can also just check for any variation, but if the CI setup ever changes i.e. due to a new version of the github actions runner image, it can break a previously green commit. My intention here is injecting the exact path we are expecting depending on the CI, but I also don't like this

Reply:
I'll actually go for this approach and remove the injection for now. If it becomes a problem, we'll have to revisit